app/src/main/java/net/bible/android/view/activity/search/MySearchResults.kt

```diff
@@ -0,0 +1,438 @@
+     * Move search results into view Adapter
+     */
+
+    private fun populateViewResultsAdapter() {
+
+        val populateViewResultsAdapterTimer = MyTimer("populateViewResultsAdapterTimer")
+        val getSearchResultVerseElementTimer = MyTimer("getSearchResultVerseElement")
+        val getSearchResultVerseStringTimer = MyTimer("getSearchResultVerseString")
+        val verseTextSpannableTimer = MyTimer("SearchHighlight")
+        val bookStatsTimer = MyTimer("Book Stats")
+        val wordStatsTimer = MyTimer("Word Stats")
+
+        mCurrentlyDisplayedSearchResults = if (isScriptureResultsCurrentlyShown) {
+            mSearchResultsHolder!!.mainSearchResults
+        } else {
+            mSearchResultsHolder!!.otherSearchResults
+        }
+        val extras = intent.extras
+        var searchDocument = extras!!.getString(SearchControl.SEARCH_DOCUMENT)
+
+        mSearchResultsArray!!.clear()
+        bookStatistics.clear()
+        wordStatistics.clear()
+
+        var listIndex = 0
+        var totalWords = 0
+        val searchHighlight = SearchHighlight(SearchControl.originalSearchString)
+        for (key in mCurrentlyDisplayedSearchResults) {
+            // Add verse to results array
+
+            var verseTextSpannable: SpannableString? = null
+            getSearchResultVerseElementTimer.start()
+            val verseTextElement = searchControl.getSearchResultVerseElement(key)
+            getSearchResultVerseElementTimer.stop()
+
+//            getSearchResultVerseStringTimer.start()
+            // This takes 3 to 4 times longer than 'getSearchResultVerseElement' which itself takes 3 to 4 times longer than the old search which is almost instant.
+            // I have converted the code to Kotlin. But apart from that it seems the same. I would like to use 'getSearchResultVerseText' for non-Strongs searches
+            // if it is faster because the Element parsing is only necessary for Strongs searches.
+            // Perhaps the performance of 'getSearchResultVerseElement' can be improved if we can work out why 'getSearchResultVerseText' is so slow.
+//            val verseTextString = searchControl.getSearchResultVerseText(key)
+//            getSearchResultVerseStringTimer.stop()
+
+
+            verseTextSpannableTimer.start()
+            verseTextSpannable = searchHighlight.generateSpannableFromVerseElement(verseTextElement)
+            verseTextSpannableTimer.stop()
+
+            mSearchResultsArray.add(SearchResultsData(listIndex, key.osisID.toString(), key.name,searchDocument, "text", verseTextSpannable!!.toHtml()))
+
+            // Add book to the book statistics array
+            bookStatsTimer.start()
+            val bookOrdinal = ((key as Verse).book as BibleBook).ordinal
+            var mBibleBook = BibleBook.values()[bookOrdinal]
+            var bookNameLong = versification.getLongName(mBibleBook)  // key.rootName
+            val bookStat = bookStatistics.firstOrNull{it.book == bookNameLong}
```

Comment:
- [ ] It looks like bookStatistics and other statistics containers could be maps, which is more efficient

Reply:
Outdated?